File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string r="";
    sort(participant.begin(),participant.end(),less<string>());
    sort(completion.begin(),completion.end(),less<string>());
    for(int i=0;i<completion.size();i++) {
        if(completion[i]!=participant[i]) {
            r+=participant[i];
            break;
        }
    }
    if(r.length()==0) {
        r+=participant[participant.size()-1];
    }
    return r;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    r=''
    participant.sort()
    completion.sort()
    for i in range(len(completion)):
        if completion[i] != participant[i]:
            r+=participant[i]
            break
    if r=='':
        r+=participant[len(participant)-1]
    return r